templates/theme-default/assets/js/components/video.js

```diff
@@ -1,66 +1,74 @@
+import isHidden from '../utils/visibility';
+
 /**
- * Initialize video player
+ * Initialize video
  *
  * @param {HTMLElement} root
  */
-const init = (root) => {
-  const video = root.querySelector('video');
-  root.classList.add('initialized');
+const init = (element) => {
+  element.dataset.video = 'initialized';
+
+  const video = element.querySelector('video');
+
+  let loaded = false;
+
+  function loadVideoSources() {
+    Array.from(video.children).forEach((videoSource) => {
+      if (typeof videoSource.tagName === 'string' && videoSource.tagName === 'SOURCE') {
+        videoSource.src = videoSource.dataset.src;
+      }
+    });
 
-  const showButton = () => root.classList.remove('is-playing');
-  const hideButton = () => root.classList.add('is-playing');
+    video.load();
+  }
 
-  if (video) {
-    video.addEventListener('playing', () => hideButton());
-    video.addEventListener('pause', () => showButton());
-    video.addEventListener('ended', () => showButton());
+  function isPlaying() {
+    return Boolean(video.currentTime > 0 && !video.paused && !video.ended && video.readyState > 2);
+  }
 
-    if (video.controls === false) {
-      root.addEventListener('click', () => {
-        if (video.paused) {
-          video.play();
-        } else {
-          video.pause();
-        }
-      });
+  function playVideo() {
+    if (!isPlaying()) {
+      video.play();
     }
+  }
 
-    if (video.paused) {
-      root.classList.remove('is-playing');
-    } else {
-      root.classList.add('is-playing');
+  function pauseVideo() {
+    if (isPlaying()) {
+      video.pause();
     }
   }
+
+  function observerCallback(entries) {
+    entries.forEach((entry) => {
+      // Break if video is hidden
+      if (isHidden()) {
+        return;
+      }
+
+      // Load sources
+      if (!loaded && entry.isIntersecting) {
+        loadVideoSources();
+        loaded = true;
+      }
+
+      // Start or pause playback depending on the visibility
+      if (entry.intersectionRatio === 0) {
+        pauseVideo();
+      } else {
+        playVideo();
+      }
+    });
+  }
+
+  // Observe the visiblity of the video element
+  const observer = new IntersectionObserver(observerCallback, { threshold: [0, 0.01] });
+  observer.observe(video);
 };
 
 export const initVideo = (root = document) => {
```

Comment:
We should call init automatically without the need to be imported & called from main.js